policy-controller/src/admission.rs

```diff
@@ -189,9 +189,13 @@ fn parse_spec<T: DeserializeOwned>(req: AdmissionRequest) -> Result<(String, Str
 
 #[async_trait::async_trait]
 impl Validate<AuthorizationPolicySpec> for Admission {
-    async fn validate(self, _ns: &str, _name: &str, spec: AuthorizationPolicySpec) -> Result<()> {
-        // TODO support namespace references?
-        if !spec.target_ref.targets_kind::<Server>() {
+    async fn validate(self, ns: &str, _name: &str, spec: AuthorizationPolicySpec) -> Result<()> {
+        if spec.target_ref.targets_kind::<Server>() {
+        } else if spec.target_ref.targets_kind::<Namespace>() {
+            if spec.target_ref.name != ns {
+                bail!("cannot target another namespace: {}", spec.target_ref.name);
+            }
+        } else {
```

Comment:
I can see it either way.  The empty if block is a bit awkward, but I think it communicates the intent well.  I also like that this expands nicely as more supported resources get added (each resource type gets an if or else-if block rather than needing an else-if with a big or containing all supported types.

Reply:
Makes sense